components/CreateReplyRequestDialog/CreateReplyRequestDialog.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+
+const localStorage = typeof window === 'undefined' ? {} : window.localStorage;
+const formInitialState = {
+  visible: false,
+  disabled: true,
+  text: '',
+};
+
+const CREATE_REPLY_REQUEST = gql`
+  mutation CreateReplyRequestFromForm($articleId: String!, $reason: String!) {
+    CreateReplyRequest(articleId: $articleId, reason: $reason) {
+      id
+    }
+  }
+`;
+const MIN_REASON_LENGTH = 80;
+
+function SubmitButton({ articleId, text, disabled, onFinish }) {
+  const [createReplyRequest] = useMutation(CREATE_REPLY_REQUEST, {
+    refetchQueries: ['LoadArticlePage'],
+  });
+
+  const handleSubmit = e => {
+    e.preventDefault(); // prevent reload
+    if (disabled) return;
+    createReplyRequest({ variables: { articleId, reason: text } });
+    onFinish();
+  };
+
+  return (
+    <button onClick={handleSubmit} disabled={disabled}>
+      {disabled ? '字數太少，無法送出' : '送出理由'}
+    </button>
+  );
+}
+
+class CreateReplyRequestDialog extends React.PureComponent {
+  static defaultProps = {};
+
+  constructor() {
+    super();
+    this.state = {
+      ...formInitialState,
+    };
+
+    this.formRef = React.createRef();
+  }
+
+  componentDidMount() {
+    const { text } = this.state;
+
+    // restore from localStorage if applicable.
+    // We don't do this in constructor to avoid server/client render mismatch.
+    //
+    this.setState({
+      text: localStorage.text || text,
+    });
+  }
+
+  set(key, value) {
+    this.setState({ [key]: value });
+
+    // Backup to localStorage
+    requestAnimationFrame(() => (localStorage[key] = value));
+  }
+
+  handleTextChange = ({ target: { value } }) => {
+    this.setState({
+      text: value,
+      disabled: !value || value.length < MIN_REASON_LENGTH,
+    });
+  };
+
+  onReasonSubmitted = () => {
```

Comment:
Let's use `handleReasonSubmit` because it is a handler method :D

We use `onXXX` for event handling props and use `handleXXX` for methods that actually implements event handling.

This helps a lot when we read code in render function, especially when we need to pass handlers from parent to children in complex controls -- if we put `handleXXX` then it means that we do some processing in the component we currently read, and if we see `<ChildComp onXXX={onXXX} />` it must be passing parent handlers `this.props.onXXX` to children.